back/routes/adventure.js

```diff
@@ -19,26 +19,40 @@ router.post('/restaurants', async(req, res) => {
   };
   axios(config)
   .then(function (response) {
-    res.status(200);
-    res.send(response.data.results);
+    res.status(200).send(response.data.results);
   })
 })
 
+// Find an user's preference
+async function preferenceQuery(username){
+  const query = new Parse.Query('UserPreference');
+  query.equalTo("username", username);
+  const currentPreference = await query.first();
+  return currentPreference.toJSON()
+}
+
 // ----- Get priorization preferences -----
 router.post('/preferences/prioritize', async(req, res) => {
-  const query = new Parse.Query('UserPreference');
-  query.equalTo("username", req.body.username);
-  let currentPreference = await query.first();
-  res.status(200).send(currentPreference.toJSON().prioritize);
+  const preference = await preferenceQuery(req.body.username)
+  res.status(200).send(preference.prioritize);
 })
 
 // ----- Get active user's preferences IDs -----
 router.post('/preferences/active', async(req, res) => {
-  const query = new Parse.Query('UserPreference');
-  query.equalTo("username", req.body.username);
-  let currentPreferences = await query.first();
-  res.status(200);
-  res.send(currentPreferences.toJSON().activePreferences);
+  const preference = await preferenceQuery(req.body.username);
+  res.status(200).send(preference.activePreferences);
+})
+
+// ----- Get active preferences' minimum values -----
+router.post('/preferences/active/values', async(req, res) => {
+  const preference = await preferenceQuery(req.body.username)
+  res.status(200).send(preference.minValues)
+})
```

Comment:
hmm, shouldn't we return this with the other endpoints? This will simplify the UI side *a lot* because we won't need to manually handle the state.